### ISSUE TITLE
source of weights in examples

### ISSUE THREAD

**Issue by abcnorio** [TimothyAlexisVass/sdxl-colorspace#1]
Hij,

(1)
in your article https://blinkingrobots.com/explaining-the-sdxl-latent-space/ related to that github repo can you please give a reference (paper...) where I can read about how you got the concrete values of

- the weights for "latents_to_rgb()"

    weights = (
        (60, -60, 25, -70),
        (60,  -5, 15, -50),
        (60,  10, -5, -35)
    )

AND

- biases_tensor:

(150, 140, 130)

Coming not from color theory I just cannot follow this part and I want to understand that from scratch how you got these values.

(2)
what does "If every worth can vary between -4 and 4 on the level of decoding, " mean exactly, I assumed after decoding we are in RGB pixel space which means values 0...255 for each pixel and each channel RGB. So "-4 ... +4" is just a linear transfo of the [0,255] space?

(3)
Again, being not that familiar with color spaces, the conversion for Cyan/Pink, Lime/Medium Purple, Sample construction) is not clear to me:

- what is exactly meant by "sample construction"? Is this what is earlier noted as "batch size"?

- a conversion for e.g. rgb(0, 255, 255)/rgb(255, 0, 0) if this is done elementwise (ie. per each channel), it would result in rgb(0,INF,INF) which is surely not meant if applied as a constant factor/ weight -> how is this applied? Assumed is for each latent pixel with its 2^14 possibilities as outlined earlier there is some kind of weighted (non-linear?) conversion possible to translate each channel to RGB space.

Thanks a lot for explanations!

best regards

**Comment by TimothyAlexisVass** [comment]
I haven't written anything on blinkingrobots, that article was taken from https://huggingface.co/blog/TimothyAlexisVass/explaining-the-sdxl-latent-space and it seems it was reworded too.

1. Someone had calculated this a long time ago and I calibrated those values. I don't know how they came about it.
2. I never wrote "if every worth" that is some nonsense blinkingrobots have changed. But to answer your question. First the data is decoded and then it's converted. The answer is a little further down in the decoding process "One key in understanding the boundary is to look at what happens in the decoding process"
3. Again, that's something blinkingrobots changed, I wrote pattern/structure.
